### PR TITLE
[Draft][circle-execution-plan] Enable execution plan for onert-micro

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR enables circle-execution-plan for onert-micro.

for issue: https://github.com/Samsung/ONE/issues/9849

draft: #9861

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 1]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
   _required_size = get_offsets_with_greedy_by_size();
+
   for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
   {
     luci::CircleNodeExecutionPlan execution_plan(i, _offsets[i]);
     luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
                              execution_plan);
   }
+
+  printf("Buffer required memory = %d\n", _required_size);
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:117]: Unintended print? or please use logging function not printf

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 2]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
+  _ordered_nodes = loco::output_nodes(_graph);
+  const auto output_required_size = get_offsets_with_greedy_by_size();
+  for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
+  {
+    luci::CircleNodeExecutionPlan execution_plan(i + input_size, _offsets[i]);
+    luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
+                             execution_plan);
+  }
+  dump_inform();
+  printf("Output graph buffer required memory = %d\n", output_required_size);
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:172]: ditto

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 3]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
+      luci::CircleNodeExecutionPlan execution_plan(
+        counter_ops + input_nodes_size + output_nodes_size, _offsets[i]);
+      luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
+                               execution_plan);
+      counter_ops++;
+    }
+  }
+
+  dump_inform();
+  printf("Buffer required memory = %d\n", _required_size);
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:261]: ditto

**Comment by seanshpark** [comment]
Title has `for onert-micro` but change is not for `onert-micro`.
It would be better to change the title to reflect the changes.

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 4]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
+}
+
 } // namespace
 
-void ExecutionPlanner::make_execution_plan()
+void ExecutionPlanner::make_execution_plan_luci_interpreter()
 {
   get_default_execution_order_plan();
   _required_size = get_offsets_with_greedy_by_size();
+
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:109]: Please split this to another PR

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp (hunk 1 above)
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:117]: ditto

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 5]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
+  const auto input_required_size = get_offsets_with_greedy_by_size();
+
+  for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
+  {
+    luci::CircleNodeExecutionPlan execution_plan(i, _offsets[i]);
+    luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
+                             execution_plan);
+  }
+  dump_inform();
+  printf("Input graph buffer required memory = %d\n", input_required_size);
```
Comment (seanshpark) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:152]: ditto

**Comment by BalyshevArtem** [comment]
> Please split this to another PR

I converted this PR as a draft and will make two PRs: for luci-interpreter and for onert-micro parts.

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp (hunk 1 above)
Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/ExecutionPlanner.cpp:118]: done

**Comment by seanshpark** [comment]
> I converted this PR as a draft and will make two PRs: for luci-interpreter and for onert-micro parts.

Thanks!
Actually, including empty line looks as formatting to enhance readability so would be not related to execution plan.

**Comment by BalyshevArtem** [comment]
everything is merged, close it
